tools/datafix/reput_bugs.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+""" Utility to reput bugs so that those with git ranges 
+    are classified with the GIT ecosystem. """
+
+from google.cloud import datastore
+
+from google.cloud import ndb
+import osv
+from google.cloud.datastore.query import PropertyFilter
+
+import argparse
+
+ndb_client = None
+MAX_BATCH_SIZE = 500
+
+
+def main() -> None:
+  parser = argparse.ArgumentParser(
+      description="Reput all bugs from a given source.")
+  parser.add_argument(
+      "--dry-run",
+      action=argparse.BooleanOptionalAction,
+      dest="dryrun",
+      default=True,
+      help="Abort before making changes")
+  parser.add_argument(
+      "--verbose",
+      action=argparse.BooleanOptionalAction,
+      dest="verbose",
+      default=False,
+      help="Display records being operated on")
+  parser.add_argument(
+      "--source",
+      action="store",
+      dest="source",
+      default="curl",
+      help="The prefix of source_id records to reput")
+  parser.add_argument(
+      "--project",
+      action="store",
+      dest="project",
+      default="oss-vdb-test",
+      help="GCP project to operate on")
+  args = parser.parse_args()
+
+  client = datastore.Client(project=args.project)
+
+  query = client.query(kind="Bug")
+  query.add_filter(filter=PropertyFilter("source", "=", args.source))
+  if not args.verbose:
+    query.keys_only()
+
+  print(f"Running query {query.filters} "
+        f"on {query.kind} (in {query.project})...")
+
+  result = list(query.fetch())
+
+  print(f"Retrieved {len(result)} bugs to examine for reputting")
+
+  # Chunk the results to reput in acceptibly sized batches for the API.
+  for batch in range(0, len(result), MAX_BATCH_SIZE):
+    try:
+      with client.transaction():
+        ndb.put_multi_async([
```

Comment:
Can you add a comment here clarifying that this will run the `_pre_put_hook()` func under Bug?

Reply:
Sure thing! Done :)